### PR TITLE
lib: lte_lc: Initialize and document event payload

### PR DESCRIPTION
**Pull request by jtguggedal**
Adds initialization of event payload structures that are copied
into the event structure.
Adds documentation of which struct member contains valid data
for each event type.

CIA-413

Signed-off-by: Jan Tore Guggedal <jantore.guggedal@nordicsemi.no>

### CODE REVIEW

File: include/modem/lte_lc.h [hunk 1]
```diff
@@ -150,26 +150,35 @@ enum lte_lc_func_mode {
 
 enum lte_lc_evt_type {
 	/** @brief Event received carrying information about the modems network registration
-	 *         status.
-	 *         Payload is of type @ref lte_lc_nw_reg_status.
+	 *         status. The event type has no associated payload.
```
Comment (tokangas) [inline include/modem/lte_lc.h:153]: Shouldn't this be the `nw_reg_status` member of type `lte_lc_nw_reg_status`?

File: include/modem/lte_lc.h [hunk 2]
```diff
@@ -150,26 +150,35 @@ enum lte_lc_func_mode {
...
 	 */
 	LTE_LC_EVT_NW_REG_STATUS,
+
 	/** @brief Event received carrying information about PSM updates. Contains PSM parameters
 	 *	   given by the network.
-	 *         Payload is of type @ref lte_lc_psm_cfg.
+	 *         The associated payload is the psm_cfg member of type @ref lte_lc_nw_reg_status
```
Comment (tokangas) [inline include/modem/lte_lc.h:159]:
```suggestion
	 *         The associated payload is the psm_cfg member of type @ref lte_lc_psm_cfg
```

**Review by tokangas — CHANGES_REQUESTED** [review]

File: include/modem/lte_lc.h [hunk 3]
```diff
@@ -150,26 +150,35 @@ enum lte_lc_func_mode {
...
+	 *	   in the event.
 	 */
 	LTE_LC_EVT_PSM_UPDATE,
+
 	/** @brief Event received carrying information about eDRX updates. Contains eDRX parameters
+	 *	   given by the network.out PSM updates. Contains PSM parameters
```
Comment (tokangas) [inline include/modem/lte_lc.h:165]: This line seems to be copy-pasted here by mistake? :)

**Review by tokangas — CHANGES_REQUESTED** [review]

File: include/modem/lte_lc.h (hunk 1 above)
Comment (jtguggedal) [reply include/modem/lte_lc.h]: It sure should, updated now

File: include/modem/lte_lc.h (hunk 3 above)
Comment (jtguggedal) [reply include/modem/lte_lc.h]: Never a PR without these copy/paste mistakes..

**Comment by jtguggedal** [comment]
@tokangas Thanks, comments have been addressed

**Review by tokangas — APPROVED** [review]
LGTM

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**